docs/source/cuda/cudapysupported.rst

```diff
@@ -0,0 +1,202 @@
+branches, which can exist as if-else and, less obvious, as for-loops.
+Another hardware limition is the number of registers, CUDA employs a large
+register table instead of a stack.  Function of high complexity may fail to
+compile due to insufficient registers.
+
+For details please consult the
+`CUDA Programming Guide
+<http://docs.nvidia.com/cuda/cuda-c-programming-guide/#programming-model>`_.
+
+Constructs
+----------
+
+The following Python constructs are not supported:
+
+* Exception handling (``try .. except``, ``try .. finally``)
+* Context management (the ``with`` statement)
+* Comprehensions (either list, dict, set or generator comprehensions)
+* Generator (any ``yield`` statements)
+
+The ``raise`` and ``assert`` statements are supported.
+See :ref:`nopython language support <pysupported-language>`.
+
+Built-in types
+===============
+
+The following built-in types support are inherited from CPU nopython mode.
+
+* int
+* float
+* complex
+* bool
+* None
+* tuple
+
+See :ref:`nopython built-in types <pysupported-builtin-types>`.
+
+
+Built-in functions
+==================
+
+The following built-in functions are supported:
+
+* :func:`abs`
+* :class:`bool`
+* :class:`complex`
+* :func:`enumerate`
+* :class:`float`
+* :class:`int`: only the one-argument form
+* :func:`len`
+* :func:`min`: only the multiple-argument form
+* :func:`max`: only the multiple-argument form
+* :class:`range`: semantics are similar to those of Python 3 even in Python 2:
+  a range object is returned instead of an array of values.
+* :func:`round`
+* :func:`zip`
+
+
+Standard library modules
+========================
+
+
+``cmath``
+---------
+
+The following functions from the :mod:`cmath` module are supported:
+
+* :func:`cmath.acos`
+* :func:`cmath.acosh`
+* :func:`cmath.asin`
+* :func:`cmath.asinh`
+* :func:`cmath.atan`
+* :func:`cmath.atanh`
+* :func:`cmath.cos`
+* :func:`cmath.cosh`
+* :func:`cmath.exp`
+* :func:`cmath.isfinite`
+* :func:`cmath.isinf`
+* :func:`cmath.isnan`
+* :func:`cmath.log`
+* :func:`cmath.log10`
+* :func:`cmath.phase`
+* :func:`cmath.polar`
+* :func:`cmath.rect`
+* :func:`cmath.sin`
+* :func:`cmath.sinh`
+* :func:`cmath.sqrt`
+* :func:`cmath.tan`
+* :func:`cmath.tanh`
+
+``math``
+--------
+
+The following functions from the :mod:`math` module are supported:
+
+* :func:`math.acos`
+* :func:`math.asin`
+* :func:`math.atan`
+* :func:`math.arctan`
+* :func:`math.acosh`
+* :func:`math.asinh`
+* :func:`math.atanh`
+* :func:`math.cos`
+* :func:`math.sin`
+* :func:`math.tan`
+* :func:`math.hypot`
+* :func:`math.cosh`
+* :func:`math.sinh`
+* :func:`math.tanh`
+* :func:`math.atan2`
+* :func:`math.exp`
+* :func:`math.expm1`             not available in Python 2.6
+* :func:`math.fabs`
+* :func:`math.log`
+* :func:`math.log10`
+* :func:`math.log1p`
+* :func:`math.sqrt`
+* :func:`math.pow`
+* :func:`math.ceil`
+* :func:`math.floor`
+* :func:`math.copysign`
+* :func:`math.fmod`
+* :func:`math.isnan`
+* :func:`math.isinf`
+
+
+``operator``
```

Comment:
The operator module appears not to be supported in CUDA at all. I tried:

```python
from numba import cuda
import numpy as np
import operator

a = np.ones(10)

@cuda.jit
def f(a, b):
    for i in range(len(a)):
        b[i] = operator.add(a[i], 2)

b = np.zeros(10)
f(a, b)
print(b)
```

Which gives:

```
$ python test4.py 
Traceback (most recent call last):
  File "/home/gmarkall/work/numba/numba/typeinfer.py", line 266, in __call__
    attrty = context.resolve_getattr(value=ty, attr=self.attr)
  File "/home/gmarkall/work/numba/numba/typing/context.py", line 101, in resolve_getattr
    attrinfo = self.attributes[value]
KeyError: Module(<module 'operator' from '/home/gmarkall/miniconda3/envs/numbadev/lib/python3.4/operator.py'>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test4.py", line 13, in <module>
    f(a, b)
  File "/home/gmarkall/work/numba/numba/cuda/compiler.py", line 504, in __call__
    kernel = self.specialize(*args)
  File "/home/gmarkall/work/numba/numba/cuda/compiler.py", line 520, in specialize
    **self.targetoptions)
  File "/home/gmarkall/work/numba/numba/cuda/compiler.py", line 49, in compile_kernel
    cres = compile_cuda(pyfunc, types.void, args, debug=debug, inline=inline)
  File "/home/gmarkall/work/numba/numba/cuda/compiler.py", line 39, in compile_cuda
    locals={})
  File "/home/gmarkall/work/numba/numba/compiler.py", line 594, in compile_extra
    return pipeline.compile_extra(func)
  File "/home/gmarkall/work/numba/numba/compiler.py", line 317, in compile_extra
    return self.compile_bytecode(bc, func_attr=self.func_attr)
  File "/home/gmarkall/work/numba/numba/compiler.py", line 326, in compile_bytecode
    return self._compile_bytecode()
  File "/home/gmarkall/work/numba/numba/compiler.py", line 581, in _compile_bytecode
    return pm.run(self.status)
  File "/home/gmarkall/work/numba/numba/compiler.py", line 209, in run
    raise patched_exception
  File "/home/gmarkall/work/numba/numba/compiler.py", line 201, in run
    res = stage()
  File "/home/gmarkall/work/numba/numba/compiler.py", line 414, in stage_nopython_frontend
    self.locals)
  File "/home/gmarkall/work/numba/numba/compiler.py", line 709, in type_inference_stage
    infer.propagate()
  File "/home/gmarkall/work/numba/numba/typeinfer.py", line 399, in propagate
    self.constrains.propagate(self.context, self.typevars)
  File "/home/gmarkall/work/numba/numba/typeinfer.py", line 98, in propagate
    constrain(context, typevars)
  File "/home/gmarkall/work/numba/numba/typeinfer.py", line 270, in __call__
    raise TypingError(msg, loc=self.inst.loc)
numba.errors.TypingError: Failed at nopython (nopython frontend)
Unknown attribute 'add' for Module(<module 'operator' from '/home/gmarkall/miniconda3/envs/numbadev/lib/python3.4/operator.py'>) $22.2 $22.3 = getattr(attr=add, value=$22.2)
File "test4.py", line 10
```